### PR TITLE
Partial fix for #8775.

### PR DESCRIPTION
**Pull request by benjamn**
This may not be a complete fix for issue #8775, but I noticed while debugging the issue that `LocalCollection._removeDollarOperators` isn't removing nested `$`-operators (as it seems like it should be): https://github.com/meteor/meteor/blob/d6c4281eebe8fffcdeb415ccc361784f854126f4/packages/minimongo/minimongo.js#L790

### CODE REVIEW

**Comment by mitar** [comment]
Are you sure it should be removing nested `$`?

**Comment by hwillson** [comment]
@mitar For `upsert`'s we should be removing nested `$` operators. You can see the reason why [here](https://github.com/meteor/meteor/blob/c2c7fab2ed59fb3087b7c6d53c643d2c184246a6/packages/minimongo/minimongo.js#L800-L801) (which was a change that came from #8631). Thanks!

**Comment by brucejo75** [comment]
Thanks for rapid debug.  Any tips on how I could patch this into my local version?  I am on windows.

#### I tried modifying files at:

> C:\Users\Bruce\AppData\Local\.meteor\packages\minimongo\1.2.1\web.browser\selector.js.

But any code changes to files there cause build errors.

#### I tried overriding `LocalCollection._removeDollarOperators `
but `LocalCollection` is not a global in the app.

#### I tried copying the package code into my packages directory and building it.

I named it in the `packages.js` file but the build could not find the package.  Which is weird because I have other local packages that work fine.

Basically, I have no idea how to patch the code without forking Meteor completely which seems like overkill.

**Comment by hwillson** [comment]
@brucejo75 - while you're waiting for this to be merged / released, you can:

1. Create a `packages` directory in your app root (if you don't have one already).
2. Copy the `meteor/meteor/packages/minimongo` directory (and contents) from the meteor `fix-issue-8775` branch, into your app's `packages` directory.
3. `meteor add minimongo`

**Comment by brucejo75** [comment]
Thanks @hwillson,

I have `METEOR_PACKAGE_DIRS` defined so I put the package in there.  That did not work.

It worked when I created another packages directory directly in my project directory.

So I am all set now.

Thanks again for the quick response!

**Comment by sebakerckhof** [comment]
@benjamn I think this introduces a new bug because you loose EJSON types.
E.g. this would now error:
```
upsert({_id: new MongoID.ObjectID("1234567890")}, {$setOnInsert: {a: 123}})
````

**Comment by hwillson** [comment]
@sebakerckhof - thanks for reporting this. Would you mind opening that up in a new issue? The changes in this PR didn't introduce that issue, as I'm able to re-create it with 1.5.

**Comment by sebakerckhof** [comment]
@hwilson, weird. Because if I revert this commit I don't get this error.

**Comment by hwillson** [comment]
Thanks @sebakerckhof - when I verified this, I tested the issue against `devel` first, then 1.5. Looks like something was still lingering from `devel` when I tested against 1.5, as I can still re-create the issue with 1.5 in my test environment. That being said, I just took a closer look at the code and see what you're saying - makes sense to me. We'll still need to track this in a new issue though since this PR has been merged. I'll move this into a new issue and work on a fix. Thanks again!
